feat: portal

DiogoKaster opened the pull request.

danielhe4rt committed:
feat(modules): initialize portal module with routes, controllers, views, and styles

danielhe4rt committed:
feat(modules): add portal module dependency in composer configuration

DiogoKaster committed:
feat: removed unused controller and created PortalPage

DiogoKaster committed:
feat: add custom layout and styling to PortalPage with the updated homepage

DiogoKaster committed:
feat: implement custom theme and fonts for Portal module with Tailwind updates

DiogoKaster committed:
feat: organize CSS with layered styles for base and theme improvements

DiogoKaster committed:
feat: enhance CSS theme management with variables and gradient background adjustments

DiogoKaster committed:
feat: add topbar styling to Portal module with Tailwind classes

DiogoKaster committed:
feat: adjust topbar padding and gradient transparency in Portal module CSS

DiogoKaster committed:
feat: add section components

DiogoKaster committed:
feat: add reusable UI components (badge, button, tag, card, headline) with accompanying styles to Portal module

DiogoKaster committed:
feat: add section component styles and update page content gradient spacing in Portal module CSS

DiogoKaster committed:
feat: add new section components and integrate hero section into homepage layout

DiogoKaster committed:
feat: implement reusable headline component with responsive styles and integrate into hero section

DiogoKaster committed:
feat: enhance headline component with badge styling and hover effect, integrate badge slot into hero section

DiogoKaster committed:
feat: add reusable button component with styles and integrate into hero section

DiogoKaster committed:
feat: add reusable avatar and avatar-stack components with styles and integrate into hero section

DiogoKaster committed:
feat: add purpose section and update hero section content in homepage layout

DiogoKaster committed:
feat: add projects section with headline component to homepage, update headline styles for improved alignment options

DiogoKaster committed:
feat: add new sections (articles, testimonials, contact) to homepage and enhance projects section with grid layout and updated styles

DiogoKaster committed:
feat: add testimonials and articles sections with headline components and responsive grid layout

DiogoKaster committed:
feat: add reusable card component with styles and integrate into purpose section

DiogoKaster committed:
feat: add reusable tag component with styles and integrate into projects section cards

DiogoKaster committed:
feat: add reusable author component with styles and integrate into articles section cards

DiogoKaster committed:
feat: enhance testimonials section with styled cards and author component, adjust card styles for consistency

DiogoKaster committed:
feat: integrate Blade Font Awesome package and reusable form components (input, textarea) into contact section styles and layout

DiogoKaster committed:
feat: update portal navigation and styles, add Blade render hooks for topbar and footer customization

DiogoKaster committed:
feat: add new logo assets, customizable footer, and improve branding styles across portal components

DiogoKaster committed:
feat: make logo component clickable and integrate as brand logo in guest panel

DiogoKaster committed:
feat: enhance hero section with background logo, interactive code component, and improve footer responsiveness

DiogoKaster committed:
feat: add contact section SVG background image asset to enhance visuals

DiogoKaster committed:
feat: add sidebar navigation Blade render hook with guest-specific buttons

DiogoKaster committed:
feat: enhance portal sections with navigation IDs, new styles, and improved ordering

DiogoKaster committed:
feat: add hover effects and label styles to sidebar items for improved interactivity

DiogoKaster committed:
Merge branch '3.x' into feat/portal

gvieira18 commented on vite.config.js:
The attribute input is duplicated instead of appending to the current array.

DiogoKaster replied:
Yup, the merging process went a bit messy, but its fixed now. Thanks

gvieira18 commented on app-modules/portal/routes/portal-routes.php:
You can delete this file if there is no route to register.

Clintonrocha98 commented on app-modules/portal/resources/css/components/headline.css:
```diff
@@ -0,0 +1,87 @@
+
+/*.hp-headline-animated {*/
+/* @apply animate-fade-in;*/
+/*}*/
```
should this be commented?

gvieira18 reviewed (changes requested):
If time is a problem, you can dismiss the changes to `portal-routes.php`, but I think the `vite.config.js` file didn't build correctly.

<img width="1162" height="551" alt="Image" src="https://github.com/user-attachments/assets/25553fbf-c001-4897-ba2e-5629a41e66da" />

Clintonrocha98 commented on config/blade-fontawesome.php:
Is it ok to leave these lines commented?

danielhe4rt commented:
> FYI: after changed to input array
> 
> <img alt="image" width="1257" height="633" src="https://private-user-images.githubusercontent.com/48625433/513454469-e35fd727-f720-4484-941b-9c3b612f697a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI5Njk0MjgsIm5iZiI6MTc2Mjk2OTEyOCwicGF0aCI6Ii80ODYyNTQzMy81MTM0NTQ0NjktZTM1ZmQ3MjctZjcyMC00NDg0LTk0MWItOWMzYjYxMmY2OTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEyVDE3Mzg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNlYzRjYWRiODg4ZWQ5YzU3MmZkOTg2MTQzYWIwNjAzNGRlMmEyZjIwNmQ0ZTcwNjc1NzliMWRkMjRiYTYzZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BDgSpw57m1piY1K7mIsglg6i_rsba2Yz-Y5NIIWz0-k">

isn't being used anymore

DiogoKaster committed:
chore: update dependencies and adjust portal styles

- Bump composer and npm dependencies, including Laravel, Symfony, Spatie, and Predis packages.
- Add `tw-animate-css` and update Vite input paths.
- Refactor portal CSS to improve imports and remove unused animations.

DiogoKaster committed:
chore: remove unused portal routes file

gvieira18 reviewed (approved):
LGTM. 👍🏻
